### PR TITLE
ci: enable stylecheck linter, fix ST1005 warnings

### PR DESCRIPTION
**Pull request by kolyshkin**
1. Enable stylecheck linter (with some checks disabled).

2. Enabled checks disabled above in golangci-extra.yml (used for new code).

3. Fix "_ST1005: error strings should not be capitalized (stylecheck)_" warnings.

### CODE REVIEW

**Review by austinvazquez — APPROVED** [review]

**Comment by AkihiroSuda** [comment]
Maybe too strict?

**Comment by kolyshkin** [comment]
> Maybe too strict?

Maybe. We wrap some errors, and those need to be non-capitalized, and we don't wrap some others, and those are actually better if capitalized.

@thaJeztah wdyt?

**Comment by kolyshkin** [comment]
Problem is, we don't know which errors are wrapped and which are not.

 - anything that is returned from libcontainer may be wrapped (so needs to be lowercased)
 - anything outside of libcontainer may or may not be wrapped
 
@AkihiroSuda what if we only apply the linter to libcontainer?

**Comment by kolyshkin** [comment]
> Problem is, we don't know which errors are wrapped and which are not.
> 
> * anything that is returned from libcontainer may be wrapped (so needs to be lowercased)
> * anything outside of libcontainer may or may not be wrapped
> 
> @AkihiroSuda what if we only apply the linter to libcontainer?

Nope; in fact, kubernetes, cri-o, containerd and other users also wrap errors printed by runc, so everything should be lowercased, so this PR makes total sense.

**Comment by AkihiroSuda** [comment]
Sorry, not sure how wrapping is relevant to capitalization

**Comment by kolyshkin** [comment]
> Sorry, not sure how wrapping is relevant to capitalization

The idea here is something like "Error: can't start container: can't run init: permission denied" looks better than "Error: Can't start container: Can't run init: permission denied". 

Here' a quote from Google Go style guide (https://google.github.io/styleguide/go/decisions#error-strings):
<blockquote>
Error strings should not be capitalized (unless beginning with an exported name, a proper noun or an acronym) and should not end with punctuation. This is because error strings usually appear within other context before being printed to the user.

```go
// Bad:
err := fmt.Errorf("Something bad happened.")
// Good:
err := fmt.Errorf("something bad happened")
```

On the other hand, the style for the full displayed message (logging, test failure, API response, or other UI) depends, but should typically be capitalized.

```go
// Good:
log.Infof("Operation aborted: %v", err)
log.Errorf("Operation aborted: %v", err)
t.Errorf("Op(%q) failed unexpectedly; err=%v", args, err)
```
</blockquote>

Another quote, from https://gist.github.com/adamveld12/c0d9f0d5f0e1fba1e551#error-strings:

> Error strings should not be capitalized (unless beginning with proper nouns or acronyms) or end with punctuation, since they are usually printed following other context. That is, use `fmt.Errorf("something bad")` not `fmt.Errorf("Something bad")`, so that `log.Print("Reading %s: %v", filename, err)` formats without a spurious capital letter mid-message. This does not apply to logging, which is implicitly line-oriented and not combined inside other messages.

File: utils_linux.go [hunk 1]
```diff
@@ -314,10 +314,10 @@ func validateProcessSpec(spec *specs.Process) error {
 		return errors.New("process property must not be empty")
 	}
 	if spec.Cwd == "" {
-		return errors.New("Cwd property must not be empty")
+		return errors.New("cwd property must not be empty")
```
Comment (thaJeztah) [inline utils_linux.go:317]: For others reviewing (I was wondering if this should be CWD), but the property is lowercase in JSON, so probably makes sense to match that https://github.com/opencontainers/runc/blob/e42446f7c68184fd1c1005f0e59a60312defb83a/vendor/github.com/opencontainers/runtime-spec/specs-go/config.go#L78-L80

**Review by thaJeztah — COMMENTED** [review]
overall I agree with these changes, but slightly concerned about changing the error-messages, as it's _possible_ there's some string-matching in other codebases that handle the errors returned from runc (binary).

**Comment by kolyshkin** [comment]
Rebased

**Comment by cyphar** [comment]
> but slightly concerned about changing the error-messages, as it's possible there's some string-matching in other codebases that handle the errors returned from runc (binary).

Maybe we should randomly change our error messages to ensure people don't do this? :wink:

**Comment by kolyshkin** [comment]
Guess I'll close this one due to lack of interest.

**Pull request closed without merging**